values.go

```diff
@@ -1929,6 +1929,23 @@ func (v Event) String() string {
 	return formatComposite(v.EventType.ID(), v.EventType.Fields, v.Fields)
 }
 
+func (v Event) GetFieldByName(fieldName string) Value {
+	for i, field := range v.EventType.Fields {
+		if field.Identifier == fieldName {
+			return v.Fields[i]
+		}
+	}
+	return nil
+}
+
+func (v Event) GetFieldsMappedByName() map[string]Value {
+	fields := make(map[string]Value, len(v.EventType.Fields))
+	for i, field := range v.EventType.Fields {
+		fields[field.Identifier] = v.Fields[i]
+	}
+	return fields
+}
```

Comment:
Nice! 

While at it, could you please also add those functions for the other composite types, i.e. `Struct`, `Resource`, `Contract`, and `Enum`? 

The current implementation could maybe be factored out into helper functions (e.g. `func GetFieldByName(values []Value, fields []Field) Value` and `func GetFieldsMappedByName(values []Value, fields []Field) map[string]Value`) and used/called in all those methods.

Reply:
what do you think about something like this?
```go

type HasFields interface {
	GetFields() []Field
	GetFieldValues() []Value
}

func GetFieldByName(v HasFields, fieldName string) Value {
	for i, field := range v.GetFields() {
		if field.Identifier == fieldName {
			return v.GetFieldValues()[i]
		}
	}
	return nil
}

func GetFieldsMappedByName(v HasFields) map[string]Value {
	fieldValues := v.GetFieldValues()
	fields := v.GetFields()
	fieldsMap := make(map[string]Value, len(fields))
	for i, field := range fields {
		fieldsMap[field.Identifier] = fieldValues[i]
	}
	return fieldsMap
}
```

then implement GetFields and GetFieldValues for those composite types?

Reply:
That also works 👍 There's a small performance hit, but it's neligable.